layers/+config-files/cfengine/packages.el

```diff
@@ -0,0 +1,51 @@
+;;; packages.el --- cfengine layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Nick Anderson <nick@cmdln.org>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;; See the Spacemacs documentation and FAQs for instructions on how to implement
+;; a new layer:
+;;
+;;   SPC h SPC layers RET
+;;
+;;
+;; Briefly, each package to be installed or configured by this layer should be
+;; added to `cfengine-packages'. Then, for each package PACKAGE:
+;;
+;; - If PACKAGE is not referenced by any other Spacemacs layer, define a
+;;   function `cfengine/init-PACKAGE' to load and initialize the package.
+
+;; - Otherwise, PACKAGE is already referenced by another Spacemacs layer, so
+;;   define the functions `cfengine/pre-init-PACKAGE' and/or
+;;   `cfengine/post-init-PACKAGE' to customize the package as it is loaded.
+
+(setq cfengine-packages
+      '(
+	flycheck
+	company
+	;; cfengine3-mode ;; This mode is included in the default emacs distro
+	))
+
+(defun cfengine/init-cfengine()
+  (use-package cfengine3-mode
+    :init
+      (add-hook 'before-save-hook 'whitespace-cleanup)
```

Comment:
The `:init` block of `use-package` should be wrapped in `progn` as per Spacemacs convention when there are more than one statements.